crates/matrix-sdk-crypto/src/machine.rs

```diff
@@ -139,14 +139,13 @@ impl OlmMachine {
     fn new_helper(
         user_id: &UserId,
         device_id: &DeviceId,
-        store: Box<dyn CryptoStore>,
+        store: Arc<dyn CryptoStore>,
         account: ReadOnlyAccount,
         user_identity: PrivateCrossSigningIdentity,
     ) -> Self {
         let user_id: Arc<UserId> = user_id.into();
         let user_identity = Arc::new(Mutex::new(user_identity));
 
-        let store: Arc<dyn CryptoStore> = store.into();
```

Comment:
> Why were all places where a CryptoStore is handled converted to use Arc instead of Box? We require them to be Sync and Send which usually means that they are already behind an Arc or something else that's easily cloneable.

It's convert to an `Arc` anyways here. This MR just stores it directly in a `Arc` so that we can have a reference to it in `BaseClient` and in `OlmMachine`